### PR TITLE
Enables the interpreter on Android

### PR DESCRIPTION
**Pull request by steveisok**
To enable on tests, you can pass `MonoForceInterpreter=true` as an extra MSBuild property.

AndroidAppBuilder now has an ForceInterpreter property which will flow the setting down
to the device / simluator.

### CODE REVIEW

File: src/mono/mono.proj [hunk 1]
```diff
@@ -252,7 +252,7 @@
       <_MonoCMakeArgs Condition="'$(Platform)' == 'arm'" Include="-DCMAKE_ANDROID_ARCH_ABI=armeabi-v7a" />
       <_MonoCMakeArgs Condition="'$(Platform)' == 'x86'" Include="-DCMAKE_ANDROID_ARCH_ABI=x86" />
       <_MonoCMakeArgs Condition="'$(Platform)' == 'x64'" Include="-DCMAKE_ANDROID_ARCH_ABI=x86_64" />
-      <_MonoCMakeArgs Include="-DENABLE_MINIMAL=ssa,portability,attach,verifier,sgen_remset,sgen_marksweep_par,sgen_marksweep_fixed,sgen_marksweep_fixed_par,sgen_copying,logging,security,shared_handles,interpreter,gac,cfgdir_config" />
+      <_MonoCMakeArgs Include="-DENABLE_MINIMAL=ssa,portability,attach,verifier,sgen_remset,sgen_marksweep_par,sgen_marksweep_fixed,sgen_marksweep_fixed_par,sgen_copying,logging,security,shared_handles,gac,cfgdir_config" />
```
Comment (akoeplinger) [inline src/mono/mono.proj:255]: do we want the interpreter to be enabled all the time? for iOS/wasm we keep it in ENABLE_MINIMAL but set `      <_MonoCMakeArgs Include="-DENABLE_INTERP_LIB=1"/>` so that the interpreter will be built as a separate library that you can use on-demand.

File: tools-local/tasks/mobile.tasks/AndroidAppBuilder/Templates/MonoRunner.java [hunk 2]
```diff
@@ -66,8 +67,8 @@ public static int initialize(String entryPointLibName, Context context) {
         // unzip libs and test files to filesDir
         unzipAssets(context, filesDir, "assets.zip");
 
-        Log.i("DOTNET", "MonoRunner initialize, entryPointLibName=" + entryPointLibName);
-        return initRuntime(filesDir, cacheDir, docsDir, entryPointLibName);
+        Log.i("DOTNET", "initRuntime, entryPointLibName=" + entryPointLibName);
```
Comment (akoeplinger) [inline tools-local/tasks/mobile.tasks/AndroidAppBuilder/Templates/MonoRunner.java:70]: unnecessary change, the old name is more descriptive

File: tools-local/tasks/mobile.tasks/AndroidAppBuilder/Templates/MonoRunner.java [hunk 3]
```diff
@@ -36,6 +36,7 @@
 
     static String entryPointLibName = "%EntryPointLibName%";
     static Bundle result = new Bundle();
+    static String isInterpreterEnabled = "%IsInterpreterEnabled%";
```
Comment (akoeplinger) [inline tools-local/tasks/mobile.tasks/AndroidAppBuilder/Templates/MonoRunner.java:39]: this should be a bool, not a string.

File: src/mono/netcore/sample/Android/AndroidSampleApp.csproj [hunk 4]
```diff
@@ -36,6 +36,7 @@
         RuntimeIdentifier="$(RuntimeIdentifier)"
         SourceDir="$(OutputPath)"
         ProjectName="HelloAndroid"
+        EnableInterpreter="true"
```
Comment (akoeplinger) [inline src/mono/netcore/sample/Android/AndroidSampleApp.csproj:39]: I wonder if we should call this `ForceInterpreter` instead since we're not just enabling the interpreter (e.g. mixed-mode style) but forcing it to be used for everything.

Comment (steveisok) [reply src/mono/netcore/sample/Android/AndroidSampleApp.csproj:39]: Should that go all the way up?  `MonoEnableInterpreter` is the MSBuild prop to kick it all off.

Comment (akoeplinger) [reply src/mono/netcore/sample/Android/AndroidSampleApp.csproj:39]: hmm it sounds clearer to me but if it means changing a lot of stuff unrelated to this then maybe it's not worth it.

File: src/mono/mono.proj (hunk 1 above)
Comment (steveisok) [reply src/mono/mono.proj:255]: When I build it as a separate library, I hit this assertion https://github.com/dotnet/runtime/blob/095adbd3503605af2723727c5dae6d7855756a8b/src/mono/mono/mini/interp-stubs.c#L153

@vargaz are there other settings that I need?

File: tools-local/tasks/mobile.tasks/AndroidAppBuilder/Templates/MonoRunner.java (hunk 3 above)
Comment (steveisok) [reply tools-local/tasks/mobile.tasks/AndroidAppBuilder/Templates/MonoRunner.java]: Call it fear of JNI ;-).

File: src/mono/mono.proj (hunk 1 above)
Comment (steveisok) [reply src/mono/mono.proj:255]: Turns out I wasn't properly initializing the interp.  I'm now doing

```
mono_jit_set_aot_mode(MONO_AOT_MODE_INTERP_ONLY);
mono_ee_interp_init ("");
```

And am getting an assertion:

```
D DOTNET  : assembly_preload_hook: System.ObjectModel  /data/user/0/net.dot.System.Buffers.Tests/files
D DOTNET  : assembly_preload_hook: System.Text.RegularExpressions  /data/user/0/net.dot.System.Buffers.Tests/files
D DOTNET  : ((null) error) * Assertion: should not be reached at /Users/steve/dev/net5/mobile-runtime/src/mono/mono/mini/tramp-amd64.c:1131
```

Comment (vargaz) [reply src/mono/mono.proj:255]: It looks like the separate interpreter lib configuration was only tested on wasm, and it might not work elsewhere.

**Review by akoeplinger — APPROVED** [review]

File: src/mono/mono.proj (hunk 1 above)
Comment (steveisok) [reply src/mono/mono.proj:255]: Issue filed https://github.com/dotnet/runtime/issues/44804
